orchestration/flows/bl832/job_controller.py

```diff
@@ -0,0 +1,332 @@
+        job_script = f"""#!/bin/bash
+#SBATCH -q debug
+#SBATCH -A als
+#SBATCH -C cpu
+#SBATCH --job-name=tomo_multires_test-0
+#SBATCH --output={scratch_path}/nerscClient-test/%x_%j.out
+#SBATCH --error={scratch_path}/nerscClient-test/%x_%j.err
+#SBATCH -N 1
+#SBATCH --ntasks-per-node 1
+#SBATCH --cpus-per-task 64
+#SBATCH --time=0:15:00
+#SBATCH --exclusive
+
+date
+srun podman-hpc run --volume {home_path}/tomo_recon_repo/microct/legacy/tiff_to_zarr.py:/alsuser/tiff_to_zarr.py \
+--volume {home_path}/tomo_recon_repo/microct/legacy/input.txt:/alsuser/input.txt \
+--volume {scratch_path}/microctdata:/alsdata \
+--volume {scratch_path}/microctdata:/alsuser/ \
+registry.nersc.gov/als/{image_name} \
+bash -c "python -m pip show ngff_zarr || python -m pip install ngff_zarr && \
+python -m pip show dask_image || python -m pip install dask_image && \
+python tiff_to_zarr.py {recon_path} --raw_file {raw_path}"
+
+date
+"""
+        try:
+            logger.info("Submitting Tiff to Zarr job script to Perlmutter.")
+            job = self.client.perlmutter.submit_job(job_script)
+            time.sleep(30)  # Wait 30 seconds before checking job completion
+            job.complete()  # waits for job completion
+            logger.info("Tiff to Zarr job completed successfully.")
+            return True
+        except Exception as e:
+            logger.error(f"Failed to submit or complete Tiff to Zarr job: {e}")
+            return False
+
+
+class HPC(Enum):
+    """
+    Enum representing different HPC environments.
+    Use enum names as strings to identify HPC sites, ensuring a standard set of values.
+
+    Members:
+        ALCF: Argonne Leadership Computing Facility
+        NERSC: National Energy Research Scientific Computing Center
+    """
+    ALCF = "ALCF"
+    NERSC = "NERSC"
+
+
+def get_controller(hpc_type: str) -> TomographyHPCController:
+    """
+    Factory function that returns an HPC controller instance for the given HPC environment.
+
+    :param hpc_type: A string identifying the HPC environment (e.g., 'ALCF', 'NERSC').
+    :return: An instance of a TomographyHPCController subclass corresponding to the given HPC environment.
+    :raises ValueError: If an invalid or unsupported HPC type is specified.
+    """
+    if not hpc_type:
+        raise ValueError("No HPC type provided.")
+
+    # Normalize input
+    hpc_str = hpc_type.strip().upper()
+
+    # Attempt to map the given string to the HPC enum
+    try:
+        hpc_enum = HPC(hpc_str)
+    except ValueError:
+        raise ValueError(f"'{hpc_type}' is not a valid HPC") from None
+
+    # Map HPC enum members to corresponding controller constructors
+    controller_map: dict[HPC, Callable[[], TomographyHPCController]] = {
+        HPC.ALCF: lambda: ALCFTomographyHPCController(),
+        HPC.NERSC: lambda: NERSCTomographyHPCController(NERSCTomographyHPCController.create_nersc_client()),
+    }
```

Comment:
Replace with if/else statements. Delay the import of the specific controller implementations. Move NERSCTomographyHPCController to nersc.py

If that doesn't work, try putting get_controller in its own method